flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/EventTimeTemporalJoinRewriteRule.java

```diff
@@ -140,7 +142,11 @@ private RelNode transmitSnapshotRequirement(RelNode node) {
             final RelNode newChild = transmitSnapshotRequirement(child);
             if (newChild != child) {
                 return wma.copy(
-                        wma.getTraitSet(), newChild, wma.rowtimeFieldIndex(), wma.watermarkExpr());
+                        wma.getTraitSet(),
+                        newChild,
+                        Collections.emptyList(),
```

Comment:
yep, added

Reply:
Ok, are all the other places where an emptyList passed into constructors ok?  

I think they may make sense as new RelNodes are being constructed.  Asking as a sanity check.

Reply:
yep, should be. fine
the hints are in `stateTtlHints` for join so should be fine
even here I guess it could be left with empty for `stateTTL`, however might be useful for other hints in future if any...